Restore TestTorchFunctionOverride

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74201
* __->__ #74202

Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

peterbell10 committed:
Restore TestTorchFunctionOverride

Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

[ghstack-poisoned]

peterbell10 committed:
Update on "Restore TestTorchFunctionOverride"


Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

[ghstack-poisoned]

peterbell10 committed:
Update on "Restore TestTorchFunctionOverride"


Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

[ghstack-poisoned]

peterbell10 committed:
Update on "Restore TestTorchFunctionOverride"


Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

[ghstack-poisoned]

peterbell10 committed:
Update on "Restore TestTorchFunctionOverride"


Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

[ghstack-poisoned]

peterbell10 committed:
Fix int-list overloading with TensorList on "Restore TestTorchFunctionOverride"


Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

[ghstack-poisoned]

peterbell10 committed:
Update on "Restore TestTorchFunctionOverride"


Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

[ghstack-poisoned]

peterbell10 committed:
Update on "Restore TestTorchFunctionOverride"


Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

[ghstack-poisoned]

peterbell10 committed:
Update on "Restore TestTorchFunctionOverride"


Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

[ghstack-poisoned]

ezyang commented:
Any word on when this will be finished? It would be helpful for these tests to be enabled for https://github.com/pytorch/pytorch/pull/75154

peterbell10 committed:
Update on "Restore TestTorchFunctionOverride"


Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

[ghstack-poisoned]

peterbell10 commented:
I've added a test skip for `histogramdd` which was the only thing blocking this so it's ready now.

ezyang commented on test/test_overrides.py:
```diff
@@ -621,6 +622,9 @@ def instance_gen():
                     func_args.append('')
+                elif t == 'SymInt':
+                    # TODO: generate actual SymbolicInt
+                    func_args.append(1)
```
This is fine, SymInt will accept regular int too. (cc @Krovatkin, but no action necessary)

ezyang commented on torch/_tensor.py:
```diff
@@ -862,6 +862,9 @@ def storage_type(self):
 
         """
+        if has_torch_function_unary(self):
+            return handle_torch_function(Tensor.storage_type, (self,), self)
```
nice

ezyang commented:
@pytorchbot merge this
